tests/utils.py

```diff
@@ -52,6 +53,23 @@ def assert_text_response(self):
             self._text == text
         ), f"Text does not match the generated completion for completion {self._completion_id}."
 
+    def assert_system_message_request(self):
+        if not self._system_message:
+            return
+
+        self.get_completion()
```

Comment:
comment on the use of `self.get_completion()`: do we only need to get the completion once? if yes, maybe update `get_completion` function to return when `self.data` and `self.response` have been set. so might be better to init both `self.data` and `self.response` in `__init__`.
another option could be calling `get_completion` in `__init__`